### PR TITLE
[ci] Split the ci/scripts directory into ci/scripts/github and ci/scripts/jenkins

### PR DESCRIPTION
**Pull request by driazati**
This is a copy of #12940 to get it running in CI. The original description is copied below:

This PR separates the files in the `ci/scripts` directory into `ci/scripts/jenkins` and `ci/scripts/github`. This is done because the `github` scripts should not be protected in the CI. #12604

### CODE REVIEW

**Comment by gigiblender** [comment]
Thank you @driazati! Cc @areusch for review.

File: ci/scripts/github/github_commenter.py [hunk 1]
```diff
@@ -18,8 +18,14 @@
 
 import re
 import logging
+import sys
+from pathlib import Path
 from typing import Dict, Tuple, Any, Optional, List, Union
 
+# Hackery to enable importing of utils from ci/scripts/jenkins
```
Comment (areusch) [inline ci/scripts/github/github_commenter.py:25]: i think at some point we were going to make this a Python package, right? is the idea that this hack bridges the gap til that happens?

File: ci/scripts/jenkins/cmd_utils.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 from typing import List
 
 
-REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent.parent
```
Comment (areusch) [inline ci/scripts/jenkins/cmd_utils.py:27]: could also
```
assert (REPO_ROOT / "Jenkinsfile").exists
```

for stability

File: tests/python/ci/test_ci.py [hunk 3]
```diff
@@ -312,7 +314,7 @@ def test_cc_reviewers(
     """
     Test that reviewers are added from 'cc @someone' messages in PRs
     """
-    reviewers_script = REPO_ROOT / "ci" / "scripts" / "github_cc_reviewers.py"
+    reviewers_script = REPO_ROOT / "ci" / "scripts" / "github" / "github_cc_reviewers.py"
```
Comment (areusch) [inline tests/python/ci/test_ci.py:317]: do we think this pattern is common enough to make `GITHUB_SCRIPT_ROOT` and `JENKINS_SCRIPT_ROOT`?

**Review by areusch — COMMENTED** [review]
thanks @driazati , left a couple comments

File: ci/scripts/github/github_commenter.py (hunk 1 above)
Comment (gigiblender) [reply ci/scripts/github/github_commenter.py:25]: Yes, I think that's needed until we create the proper package structure.

**Pull request closed without merging**